### PR TITLE
Ask for confirmation when removing opam packages or switches

### PR DESCRIPTION
**Pull request by mnxn**
Removing a switch or package is currently done without a confirmation box. Users could accidentally uninstall a package or switch and have to wait a long time to reinstall everything they need.

A simple confirmation dialog (using `showInformationMessage`) should be sufficient to avoid accidental removals.

### CODE REVIEW

File: src/import.ml [hunk 1]
```diff
@@ -68,3 +68,12 @@ let open_file_in_text_editor target_uri =
   in
   let+ text_editor = Window.showTextDocument ~document:(`TextDocument doc) () in
   text_editor
+
+let ask_confirmation message =
+  let open Promise.Syntax in
+  let+ choice =
+    Vscode.Window.showInformationMessage ~message
+      ~choices:[ ("Yes", true); ("No", false) ]
```
Comment (tmattio) [inline src/import.ml:76]: I would make the actions more explicit:  e.g. "Remove switch" "Cancel"

File: src/treeview_switches.ml [hunk 2]
```diff
@@ -101,22 +101,31 @@ module Command = struct
...
-              Vscode.Commands.executeCommand
-                ~command:Extension_consts.Commands.refresh_switches ~args:[]
-            in
-            let (_ : Ojs.t option Promise.t) =
-              Vscode.Commands.executeCommand
-                ~command:Extension_consts.Commands.refresh_sandbox ~args:[]
-            in
-            show_message `Info "The switch has been removed successfully." )
+          let* confirmed =
+            ask_confirmation
```
Comment (tmattio) [inline src/treeview_switches.ml:107]: What do you think of a function `with_confirmation` that takes a `unit -> unit Promise.t` and executes it only if the user confirmed it?

**Review by tmattio — APPROVED** [review]
LGTM, apart from my comment on having more explicit action buttons

File: src/treeview_switches.ml (hunk 2 above)
Comment (mnxn) [reply src/treeview_switches.ml:107]: Sounds good.

**Pull request merged**